packages/portfolio-contract/src/pos-gmp.flows.ts

```diff
@@ -296,3 +296,51 @@ export const CompoundProtocol = {
     await sendGMPContractCall(target, calls, fee, lca, gmpChain);
   },
 } as const satisfies ProtocolDetail<'Compound', AxelarChain, EVMContext>;
+
+/**
+ * see {@link https://github.com/beefyfinance/beefy-contracts/blob/master/contracts/BIFI/vaults/BeefyVaultV7.sol }
+ * 0878a68 Nov 2022
+ */
+type BeefyVaultI = {
+  deposit: ['uint256'];
+  withdraw: ['uint256'];
+};
+
+const BeefyVault: BeefyVaultI = {
+  deposit: ['uint256'],
+  withdraw: ['uint256'],
+};
+
+export const BeefyProtocol = vaultName =>
+  ({
+    protocol: 'Beefy',
+    chains: keys(AxelarChain) as AxelarChain[],
+    supply: async (ctx, amount, src) => {
+      const { addresses: a, lca, gmpChain, gmpFee: fee } = ctx;
+      const session = makeEVMSession();
+      const usdc = session.makeContract(a.usdc, ERC20);
+      const vaultAddress = a[`beefy_${vaultName}`];
+      const vault = session.makeContract(vaultAddress, BeefyVault);
+      usdc.approve(vaultAddress, amount.value);
+      vault.deposit(amount.value);
+      const calls = session.finish();
+
+      const { chainName, remoteAddress } = src;
+      const axelarId = ctx.axelarIds[chainName];
+      const target = { axelarId, remoteAddress };
+      await sendGMPContractCall(target, calls, fee, lca, gmpChain);
+    },
+    withdraw: async (ctx, amount, dest) => {
+      const { addresses: a, lca, gmpChain, gmpFee: fee } = ctx;
+      const session = makeEVMSession();
+      const vaultAddress = a[`beefy_${vaultName}`];
+      const vault = session.makeContract(vaultAddress, BeefyVault);
+      vault.withdraw(amount.value);
+      const calls = session.finish();
+
+      const { chainName, remoteAddress } = dest;
+      const axelarId = ctx.axelarIds[chainName];
+      const target = { axelarId, remoteAddress };
+      await sendGMPContractCall(target, calls, fee, lca, gmpChain);
+    },
+  }) as const satisfies ProtocolDetail<'Beefy', AxelarChain, EVMContext>;
```

Comment:
```suggestion
  }) as const satisfies ProtocolDetail<'Beefy', AxelarChain, EVMContext & { vaultName: BeefyVault }>;
```

(presuming `BeefyVault` is right)